services/meta/client.go

```diff
@@ -400,6 +400,37 @@ func (c *Client) User(name string) (*UserInfo, error) {
 	return nil, ErrUserNotFound
 }
 
+// UserWithPassword returns a user if the given password is correct.
+func (c *Client) UserWithPassword(username, password string) (*UserInfo, error) {
```

Comment:
I don't really want to add another method if possible, so I think it would be better to just call `User` and then try to call `Authenticate` with the username and check the result. That way we can avoid a second function that needs to be maintained. You can also do something like this:

```
u, err := e.MetaClient.Authenticate(user, password)
if err == ErrUserNotFound {
    return err
} else if u == nil {
    u, err = e.MetaClient.User(user)
    if err != nil {
        return err
   }
}
// do stuff with the user
```